### DISCUSSION TITLE
How to get an oracle price quote on devnet (agoricdev-11)

### DISCUSSION THREAD

**Discussion by dckc** [Agoric/agoric-sdk#7783] [Q&A] [answered]
Once you have your [solo wallet client](https://agoric.com/documentation/getting-started/beta.html#set-up-a-wallet-agoric-solo-machine) set up with the [REPL](https://agoric.com/documentation/getting-started/start-a-project.html#deploy-the-contract-and-api) open:

Your `home` object has a `priceAuthority` property:

```
command[0] pa = home.priceAuthority
history[0] [Object Alleged: fake price authority]{}
```

Next you need the brands of the tokens in which the oracle issues quotes.
I tend to put each brand on a "kit" object for that token:

```
command[1] lookup('agoricNames', 'oracleBrand', 'USD').then(brand => (USD={brand}))
history[1] {"brand":[Object Alleged: USD brand]{}}
command[2] lookup('agoricNames', 'oracleBrand', 'ATOM').then(brand => (ATOM={brand}))
history[2] {"brand":[Object Alleged: ATOM brand]{}}
```

If we add a `unit` property to our token kits, we can use a `mk` function to abbreviate ERTP amounts:

```
command[4] USD.unit = ATOM.unit = 1_000_000n
history[4] 1000000n
command[7] mk = (kit, qty) => ({brand: kit.brand, value: qty * kit.unit})
history[7] [Function mk]
```

Now we're ready to `makeQuoteNotifier` for 1,000 ATOM in USD:

```
command[13] E(pa).makeQuoteNotifier(mk(ATOM, 1_000n), USD.brand)
history[13] [Object Alleged: QuoteNotifier]{}
command[14] qn=history[13]
history[14] [Object Alleged: QuoteNotifier]{}
```

And finally, as with all [notifiers](https://agoric.com/documentation/guides/js-programming/notifiers.html),  we can `getUpdateSince` to get an update; that is: a quote:

```
command[15] E(qn).getUpdateSince()
history[15] {"updateCount":107,"value":{"quoteAmount":{"brand":[Object Alleged: quote brand]{},"value":[{"amountIn":{"brand":[Object Alleged: ATOM brand]{},"value":1001000n},"amountOut":{"brand":[Object Alleged: USD brand]{},"value":14225711n},"timer":[Object Alleged: timerService]{},"timestamp":1652215788n}]},"quotePayment":[Object Alleged: quote payment]{}}}
```

We can continue to get quotes on demand:

```
command[16] E(qn).getUpdateSince(107)
history[16] {"updateCount":108,"value":{"quoteAmount":{"brand":[Object Alleged: quote brand]{},"value":[{"amountIn":{"brand":[Object Alleged: ATOM brand]{},"value":1001000n},"amountOut":{"brand":[Object Alleged: USD brand]{},"value":14179753n},"timer":[Object Alleged: timerService]{},"timestamp":1652215801n}]},"quotePayment":[Object Alleged: quote payment]{}}}
command[17] E(qn).getUpdateSince(108)
history[17] {"updateCount":123,"value":{"quoteAmount":{"brand":[Object Alleged: quote brand]{},"value":[{"amountIn":{"brand":[Object Alleged: ATOM brand]{},"value":1001000n},"amountOut":{"brand":[Object Alleged: USD brand]{},"value":14014448n},"timer":[Object Alleged: timerService]{},"timestamp":1652216274n}]},"quotePayment":[Object Alleged: quote payment]{}}}
```

That's kind of a mouthful:

 - The update has `updateCount` and `value` properties.
    - The value has `quoteAmount` and `quotePayment` properties.
      - The `quoteAmount`, like all ERTP amounts, has a `brand` and a `value`.
         - The `value` has `amountIn` and `amountOut` (as well as `timer` and `timestamp`)
            - The `value`s of `amountIn` and `amountOut` are in the smallest denomination, 10<sup>-6</sup> ATOM / USD resp.

So let's make a little function to format quote amounts:

```
command[36] fmtQuote6 = ({value: [{amountIn, amountOut}]}) => `${(Number(amountOut.value) / Number(amountIn.value)).toFixed(3)} ${amountIn.brand} / ${amountOut.brand}`.replace(/\[object Alleged: (\w+) brand\]/g, '$1')
history[36] [Function fmtQuote6]
```

Now when `getUpdateSince` gives us all that stuff:

```
command[38] E(qn).getUpdateSince(123)
history[38] {"updateCount":224,"value":{"quoteAmount":{"brand":[Object Alleged: quote brand]{},"value":[{"amountIn":{"brand":[Object Alleged: ATOM brand]{},"value":1001000n},"amountOut":{"brand":[Object Alleged: USD brand]{},"value":13374841n},"timer":[Object Alleged: timerService]{},"timestamp":1652219283n}]},"quotePayment":[Object Alleged: quote payment]{}}}
```

we can pick out just the `quoteAmount`:

```
command[39] ({value: { quoteAmount }} = history[38]), null
history[39] null
```

and then format it nicely:

```
command[40] fmtQuote6(quoteAmount)
history[40] "13.361 ATOM / USD"
```

### Screenshot

![image](https://user-images.githubusercontent.com/150986/167722617-a84e090b-7800-4840-b791-120c8dda3a5c.png)

### Governance proposal to choose the oracle operator for agoricdev-11

https://devnet.explorer.agoric.net/proposals/3

**Comment by Tartuffo** [comment]
@btulloh here ya go!

**Comment by dckc** [comment, marked as answer]
I neglected to split the question and the answer, but this is answered.
